R/legacy-format-support.R

```diff
@@ -1,4 +1,4 @@
-map_old_format <- function(df) {
+map_old_format <- function(df, filename=NULL) {
   if ("ensemble" %in% colnames(df)) {
     df <- df |>
       dplyr::mutate(family = "sample") |> 
```

Comment:
Should the family be "ensemble" per our last standards call?

Reply:
Users can still use the term `ensemble`, this line of code will ensure it gets handled correctly.  We are going to have to change the word to "sample" somewhere in the code or the code isn't going to work -- we are calling functions from the distributional package down the road and those functions want the word to be `sample`.  

The current setup does mean that it becomes the word "sample" in the output, i.e. the family column of the scoring table.  We could avoid that by changing all the words "sample" back to "ensemble" as an additional step before writing out the scores, but that really feels like slowing down the code for no reason.  


If we really can't get ecologists to understand that "ensemble" and "sample" mean the same thing, I think we aren't going to convince them that "ensemble distributions" and "parametric distributions" can both be handled in a common framework (i.e. using our three columns of `family`, `parameter`, `prediction`).  

Also most ecologists are just not familiar with the word "ensemble" -- it's pretty niche to modelers.  IMO, most ecologists get at least one stats course and are familiar with stats terms like "sample", but only those that go into modeling learn about ensembles, so "sample" feels like a sensible default.  (i.e. the audience that is more burdened by that default is arguably also the one with the greater quantitative background and thus better prepared to realize  these are mathematically equivalent notions...)

Reply:
While I don't disagree, my comment was based on the outcome of the meeting this week.  It seemed that folks agreed that it was ok to have ensemble as the family so forecasts are going to come in with ensemble as the family and we need to handle that.  "Also most ecologists are just not familiar with the word ensemble" is the exact opposite of Mike's statement in the meeting so there seems to be issues still to iron out.  Can we convert a family `ensemble` to `sample` here as well?

Reply:
Summarizing: 

1. in this PR, we still accept both `ensemble` and `sample` as valid "family" inputs
2. but we internally convert everything so that the functions can actually work
3. we also 'standardize' the family term in the output 'score' data.frame, so that it follows distributional standard

Arguably we could still do #2 but change #3 to 'standardize' on the word "ensemble" in the output scores.  We could be more aggressive still and force incoming forecasts to only use the word "ensemble" and reject family being "sample" -- this would avoid a user finding their family had changed (i.e. from sample into ensemble) in the output scores, (not changing it would be worse I think as the scores files would be using two terms to describe the same thing!).  But rejecting "sample" as a valid family while supporting all other distributional names and forcing everyone to use the word ensemble feels heavy-handed to me, given that our justification is "don't confuse people" and most ecologists are more familiar with the idea of 'sample' than the idea of 'ensemble'....

Reply:
Doesn't it only support "ensemble" as a column, not a value in the family column?  I am saying maybe add 
```
forcats::fct_recode(family, sample="ensemble")
``